### ISSUE TITLE
i have very high mAp

### ISSUE THREAD

**Issue by moahaimen** [ultralytics/yolov5#9325]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

@glenn-jocher 
hi I made a custom dataset with very high mAP Results in training
![image](https://user-images.githubusercontent.com/30658799/189020115-e57ac00c-37f8-4446-85d7-d920493903e9.png)

but when I make detection on a video
the detection sometimes gives very high confidence but more often gives very low conf.
also, it makes many mistakes between objects why?

**Comment by MartinPedersenpp** [comment]
How big is your dataset?
Are you training and validating on the same images/labels?
Have you tried evolving the hyperparameters? Perhaps your classes are rotated compared to the training material.

**Comment by moahaimen** [comment]
"Have you tried evolving the hyperparameters? Perhaps your classes are rotated compared to the training material."
Could you explain more please

**Comment by MartinPedersenpp** [comment]
> "Have you tried evolving the hyperparameters? Perhaps your classes are rotated compared to the training material." Could you explain more please

https://docs.ultralytics.com/yolov5/tutorials/hyperparameter_evolution

**Comment by MartinPedersenpp** [comment]
> @MartinPedersenpp I made a code that generates objects 7 objects in each image of the 30000 training images, and also it preprocessed and augmented, and then I annotate each object to its class and as you see above, the training results are very high

I am not sure what you mean by generating objects, are you sure that your test images are similar to your train/val images?

**Comment by MartinPedersenpp** [comment]
Hmm I think you need to check your dataset / annotations.
As far as I understand, a high MaP comes from the validation prediction matching your validation annotations, so as far as I can understand, your model is performing well when validating, but not when you test.
Have you tried taking some random samples from your train / val folders and draw the bboxes from the annotations, just to check that they are correct?
I once converted annotations from openimages to yolo format, but I had calculated them incorrectly, so the bboxes where way off.

**Comment by sarpx** [comment]
The mPAs you get at the end of the training show that the model works well at the end of the training. But your dataset may not be representative of the space you are testing. Examine all the images with low results and make sure you have training data that represents this environment.

**Comment by moahaimen** [comment]
> The mPAs you get at the end of the training show that the model works well at the end of the training. But your dataset may not be representative of the space you are testing. Examine all the images with low results and make sure you have training data that represents this environment.

thank you for advising i will certainly do that

**Comment by glenn-jocher** [comment]
@moahaimen you're welcome! If you need any further assistance with optimizing your training or dataset representation, feel free to ask. Good luck!
